sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -22,7 +22,7 @@ export interface ContextAPI {
 // @public
 export function createSpanFunction(args: CreateSpanFunctionArgs): <T extends {
     tracingOptions?: OperationTracingOptions | undefined;
-}>(operationName: string, operationOptions: T | undefined) => {
+}>(operationName: string, operationOptions?: T | undefined, startSpanOptions?: SpanOptions | undefined) => {
```

Comment:
Will you consider changing the type of `SpanOptions` to a union of strings in another PR?

Reply:
It's where I want to get to eventually but at least this puts us out of the business of carrying enum types around in option bags. An enum as a function param feels less painful than an enum as a public API member.

What I think, after our discussion, is we need to do a few things before we can do that:

1. Offer higher-level abstractions that allows tracing without worrying about mechanics (spanType, closing spans, etc) of interacting with OTel spans.
2. Decouple OTel's Span from our own

We'd have to think of what API makes sense, and I suspect that will take some time. I can _probably_ make SpanKind a string union before GA and do the conversion in `createTrace` but that won't solve for SpanStatusCode.

Here's why we can't do that today: https://github.com/Azure/azure-sdk-for-js/blob/b7e4ab869ad03208f4502c913e83a36c06b03009/sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts#L513-L516 as an example - everyone gets back an OTel span and sets the status on it. We'd have to get away from that with something like the `withTrace` pattern I showed. I don't think we'll get there in time with upgrading all the libraries to use any new patterns before GA.

But, definitely open to suggestions and ideas to do this without too much code churn in client libraries!